Multisig/App/SceneDelegate.swift

```diff
@@ -83,14 +113,49 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
     func sceneDidEnterBackground(_ scene: UIScene) {
         // Save changes in the application's managed object context when the application transitions to the background.
         App.shared.coreDataStack.saveContext()
+        SynchronizationService.shared.stopSyncLoop()
     }
 
     func sceneDidDisconnect(_ scene: UIScene) {
         // do nothing
     }
 
+    #warning("TODO: finish deep links support for WalletConnect")
+    func scene(_ scene: UIScene, openURLContexts URLContexts: Set<UIOpenURLContext>) {
+        for context in URLContexts {
+            print("[WALLETCONNECT]")
+            print("url: \(context.url.absoluteURL)")
+            print("scheme: \(String(describing: context.url.scheme))")
+            print("host: \(String(describing: context.url.host))")
+            print("path: \(context.url.path)")
+            print("components: \(context.url.pathComponents)")
+        }
+    }
+
+    func scene(_ scene: UIScene, continue userActivity: NSUserActivity) {
+        // Get URL components from the incoming user activity.
+        guard userActivity.activityType == NSUserActivityTypeBrowsingWeb,
+              let incomingURL = userActivity.webpageURL,
+              let components = NSURLComponents(url: incomingURL, resolvingAgainstBaseURL: true) else {
+            return
+        }
+
+        // Check if URL is a potential WalletConnect session
+        guard let params = components.queryItems,
+              !params.isEmpty,
+              let wcURL = params[0].value else {
+            return
+        }
+
+        try? WalletConnectServerController.shared.connect(url: wcURL)
+    }
+
     // MARK: - Window Management
 
+    func present(_ controller: UIViewController) {
+        tabBarWindow?.rootViewController?.present(controller, animated: true)
```

Comment:
check with the ViewControllerFactory to see if method can be reused to open the screen

Reply:
I forgot what did we discuss here

Reply:
added to TODO list